cli/cmd/tapRunner.go

```diff
@@ -252,77 +251,97 @@ func updateMizuTappers(ctx context.Context, kubernetesProvider *kubernetes.Provi
-		if !config.Config.IsNsRestrictedMode() {
-			if err := kubernetesProvider.RemoveNonNamespacedResources(removalCtx, mizu.ClusterRoleName, mizu.ClusterRoleBindingName); err != nil {
-				logger.Log.Errorf(uiUtils.Error, fmt.Sprintf("Error removing non-namespaced resources: %v", errormessage.FormatError(err)))
-				return
-			}
-		} else {
-			if err := kubernetesProvider.RemoveServicAccount(removalCtx, config.Config.MizuResourcesNamespace, mizu.ServiceAccountName); err != nil {
-				logger.Log.Errorf(uiUtils.Error, fmt.Sprintf("Error removing Service Account %s in namespace %s: %v", mizu.ServiceAccountName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
-				return
-			}
+	if err := kubernetesProvider.RemoveNonNamespacedResources(ctx, mizu.ClusterRoleName, mizu.ClusterRoleBindingName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing non-namespaced resources: %v", errormessage.FormatError(err)))
+	}
 
-			if err := kubernetesProvider.RemoveRole(removalCtx, config.Config.MizuResourcesNamespace, mizu.RoleName); err != nil {
-				logger.Log.Errorf(uiUtils.Error, fmt.Sprintf("Error removing Role %s in namespace %s: %v", mizu.RoleName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
-			}
+	if err := kubernetesProvider.RemoveServicAccount(ctx, config.Config.MizuResourcesNamespace, mizu.ServiceAccountName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing Service Account %s in namespace %s: %v", mizu.ServiceAccountName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
+	}
 
-			if err := kubernetesProvider.RemoveRoleBinding(removalCtx, config.Config.MizuResourcesNamespace, mizu.RoleBindingName); err != nil {
-				logger.Log.Errorf(uiUtils.Error, fmt.Sprintf("Error removing RoleBinding %s in namespace %s: %v", mizu.RoleBindingName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
-			}
-		}
+	if err := kubernetesProvider.RemoveRole(ctx, config.Config.MizuResourcesNamespace, mizu.RoleName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing Role %s in namespace %s: %v", mizu.RoleName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
 	}
 
-	if !config.Config.IsNsRestrictedMode() {
-		waitUntilNamespaceDeleted(removalCtx, cancel, kubernetesProvider)
+	if err := kubernetesProvider.RemoveRoleBinding(ctx, config.Config.MizuResourcesNamespace, mizu.RoleBindingName); err != nil && !isErrForbiddenGet(err) {
+		level := getLogLevel(isErrForbiddenGet(err))
+		logger.Logf(level, uiUtils.Error, fmt.Sprintf("Error removing RoleBinding %s in namespace %s: %v", mizu.RoleBindingName, config.Config.MizuResourcesNamespace, errormessage.FormatError(err)))
```

Comment:
why do we need "uiUtils.Error" if it will be printed to the file only "debug" (will look strange with additional chars)

Reply:
1. Where do you see additional chars? When I open the log file I see colors, not control chars.
2. We already print colors to the debug log. If this is an issue, then maybe we should write a new Log.Debug method that filters out non ASCII chars or control chars? [not in this PR]
3. I think it would be better to have `Log.Errorf` add the color, and not have `uiUtils.Error`spread all over the code. I'm not sure how to do this yet, and anyway it's for this PR.
4. I agree that it would be weird to have red message that start with the word "Error" but in DEBUG log level. On the other hand I tried to write generic reusable code.